File: frontend/src/config/dev-config.ts
// Development configuration to bypass authentication
const isDevelopment = 
  import.meta.env.DEV || 
  window.location.hostname === 'localhost' ||
  window.location.hostname === '127.0.0.1';

export const DEV_CONFIG = {
  // Set to true to bypass all authentication (auto-enabled in development)
  BYPASS_AUTH: isDevelopment,
  
  // Mock user data for development
  MOCK_USER: {
    id: 1,
    email: 'dev@example.com',
    full_name: 'Development User',
    is_active: true,
    is_superuser: true,
    created_at: new Date().toISOString(),
    updated_at: new Date().toISOString()
  },
  
  // Mock token for API requests
  MOCK_TOKEN: 'mock-dev-token-for-development-only'
};

File: backend/app/core/dev_config.py
# app/core/dev_config.py
"""
Development configuration to enable/disable authentication
Import this in auth-related functionality to bypass auth checks in development
"""
import os

# Auto-detect development environment
is_development = os.getenv("DEBUG") == "1" or os.getenv("ENVIRONMENT") == "development"

# Set to True to bypass authentication checks
BYPASS_AUTH = is_development

# Mock user data for development when auth is bypassed
MOCK_USER = {
    "id": 1,
    "email": "dev@example.com",
    "full_name": "Development User",
    "is_active": True,
    "is_superuser": True,
    "created_at": None,  # Will be set by DB if user needs to be created
    "updated_at": None   # Will be set by DB if user needs to be created
}